src/api/spec/models/kiwi/image_spec.rb

```diff
@@ -76,4 +82,60 @@
       it { expect(subject.valid?).to be_falsey }
     end
   end
+
+  describe '.to_xml' do
+    context 'without kiwi image file' do
+      it 'returns nil' do
+        dbl_package = double("Some Package")
+        allow(dbl_package).to receive(:kiwi_image_file)
+        allow(kiwi_image).to receive(:package).and_return(dbl_package)
+
+        expect(kiwi_image.to_xml).to be_nil
+      end
+    end
+
+    context 'with kiwi image file' do
+      before do
+        dbl_package = double('Some Package')
+        allow(dbl_package).to receive(:kiwi_image_file).and_return('fake_filename.kiwi')
+        allow(dbl_package).to receive_messages(source_file: kiwi_xml)
+        allow(kiwi_image).to receive(:package).and_return(dbl_package)
+      end
+
+      subject { Nokogiri::XML::DocumentFragment.parse(kiwi_image.to_xml) }
+
+      it { expect(subject.errors).to be_empty }
+      it { expect(subject.xpath('.//image').length).to be(1) }
+      it { expect(subject.xpath('.//image/description').length).to be(1) }
+      it { expect(subject.xpath('.//image/packages/package').length).to be(20) }
+      it { expect(subject.xpath('.//image/repository').length).to be(0) }
+    end
+  end
+
+  describe '.write_to_backend' do
+    let(:user) { create(:confirmed_user, login: 'tom') }
+    let(:project) { create(:project, name: 'fake_project') }
+    let(:package) do
+      create(:package_with_kiwi_file,
+             name: 'package_with_valid_kiwi_file', project: user.home_project, kiwi_image_id: kiwi_image.id, kiwi_file_content: kiwi_xml)
+    end
+
+    before do
+      login user
+    end
+
+    context 'without any exception' do
+      subject { package.kiwi_image.write_to_backend }
+
+      it { expect(subject).to be_truthy }
+    end
+
+    context 'with exception' do
+      before do
+        allow_any_instance_of(Kiwi::Image).to receive(:save!).and_raise(APIException)
+      end
+
+      it { expect{ package.kiwi_image.write_to_backend }.to raise_error(APIException) }
+    end
```

Comment:
I think what you are testing in the first `context` is that `package.save_file` works (which is already tested somewhere else I suppose) and what you are testing in the second `context` is that `rescue` works which is already tested in the rails test suite ;-)